compiler/compile/OMRCompilation.cpp

```diff
@@ -2165,6 +2165,10 @@ OMR::Compilation::notYetRunMeansCold()
 void
 OMR::Compilation::setReturnInfo(TR_ReturnInfo i)
    {
+#ifndef TR_TARGET_64BIT
+   TR_ASSERT_FATAL(i != TR_ObjectReturn, "TR_ObjectReturn should not be used for 32bit architecture");
```

Comment:
Can you explain this restriction please?  Of the top of my head I'm not aware of a reason why it couldn't be used...

Reply:
The comment on `TR_ObjectReturn` says:
https://github.com/eclipse/omr/blob/26b4ee8476bcbab916c7cf8a9251ea5ae80b260f/compiler/compile/CompilationTypes.hpp#L93-L95.
I'm just being defensive. If this is not true or desirable, I'm happy to remove the assert (to be honest, I was not sure whether or not include it but then I decided to see what other would think)

Reply:
I'm not sure why there would be a bitness restriction on its usage.  Z seems to use it even for 32-bit.  https://github.com/eclipse/omr/blob/26b4ee8476bcbab916c7cf8a9251ea5ae80b260f/compiler/z/codegen/ControlFlowEvaluator.cpp#L696

Please remove the assert.